query-engine/prisma-models/src/fields.rs

```diff
@@ -144,18 +98,21 @@ impl Fields {
             .collect()
     }
 
-    pub fn find_many_from_relation(&self, names: &BTreeSet<String>) -> Vec<RelationFieldRef> {
-        self.relation_weak()
-            .iter()
-            .filter(|field| names.contains(&field.upgrade().unwrap().name))
-            .map(|field| field.upgrade().unwrap())
-            .collect()
-    }
-
-    pub fn find_from_all(&self, prisma_name: &str) -> crate::Result<&Field> {
+    pub fn find_from_all(&self, prisma_name: &str) -> crate::Result<Field> {
         self.all
             .iter()
             .find(|field| field.name() == prisma_name)
+            .map(Clone::clone)
+            .or_else(|| {
+                let model = self.model();
+                let internal_data_model = model.internal_data_model();
+                let id = internal_data_model
+                    .walk(model.id)
+                    .relation_fields()
+                    .find(|rf| rf.name() == prisma_name)
+                    .map(|rf| rf.id);
+                id.map(|id| Field::from(internal_data_model.clone().zip(id)))
+            })
```

Comment:
This is temporarily very ugly, because we have a hybrid of local scalar and composite fields, and relation fields that are ported to the new system.